### PR TITLE
💄Menu: focus-ring from :focus to :focus-visible

### LINKED ISSUE #3394: Focus state on click in menu
**Describe the bug**

For dropdown menu, the focus border apper on click. my thoughts is that hover effect, and active sign gives enough signal. So i am suggesting that Focus border is only shown for keybord navigation. 

image

**Extra**
Posted about this in the slack channel and a answer was: 

The focus state is important for keyboard navigation. In Menu for some reason it is also visible on click. I am not sure if that was a conscious design/ux decision back in the day or just an oversight in the implementation, because it uses :focus instead of :focus-visible which is how it is done everywhere else (only show up during keyboard navigation). I recommend filing a bug ticket so we can decide if we should change this :+1:

### PR DESCRIPTION
**Pull request by oddvernes**
resolves #3394 

I believe this was an oversight when making the component and not intended interaction design

### CODE REVIEW

File: packages/eds-core-react/src/components/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -61,9 +61,6 @@ const Item = styled.button.attrs<StyleAttrsProps>(({ $isFocused }) => ({
           svg {
             fill: ${icon.states.disabled.typography.color};
           }
-          &:focus {
```
Comment (torleifhalseth) [inline packages/eds-core-react/src/components/Menu/MenuItem.tsx:64]: Will this get the default browser focus now?

File: packages/eds-core-react/src/components/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -78,7 +75,7 @@ const Item = styled.button.attrs<StyleAttrsProps>(({ $isFocused }) => ({
               background: ${hover.background};
             }
           }
-          &:focus {
+          &:focus-visible {
             z-index: 3;
```
Comment (torleifhalseth) [inline packages/eds-core-react/src/components/Menu/MenuItem.tsx:79]: Do you know why we need this?

File: packages/eds-core-react/src/components/Menu/MenuItem.tsx (hunk 1 above)
Comment (oddvernes) [reply packages/eds-core-react/src/components/Menu/MenuItem.tsx:64]: nothing happens on focus now, only when focusing using keyboard (`focus-visible`) where it gets the same moss-green dashed focus as our other components.

File: packages/eds-core-react/src/components/Menu/MenuItem.tsx (hunk 2 above)
Comment (oddvernes) [reply packages/eds-core-react/src/components/Menu/MenuItem.tsx:78]: This is needed so the focus.ring does not clip below the next item. However the z-index: 1 above in hover is no longer needed so i removed that one

**Review by torleifhalseth — APPROVED** [review]
LGTM 💅

**Pull request merged**